pvm.py

```diff
@@ -36,29 +37,90 @@ def _init_logger():
 VIDEO_PATH = "jellyfish720p.mp4"
 media = ""
 IS_FILE_SET = False
+NEXT_TIME = None
+
+'''
+UDP command example:
+/PVM 12:13:14
+/PVM <action_name> <action_value>
+
+The normal workflow is below,
+1. Receive the first command: /PVM 12:13:14
+	- parse the time
+	- plus 3s
+	- save the time into NEXT_TIME
+2. Receive the second command: /PVM file jellyfish.mp4
+	- Wait until the NEXT_TIME
+	- Execute the command
+	- Set NEXT_TIME = None
+'''
 
 def parse_commands(*args):
 	global media
 	global VIDEO_PATH
 	global IS_FILE_SET
+	global NEXT_TIME
+	# Parse the time send in UDP, 
+	# From /PVM 12 13 14 to 12:13:14
+	if len(args) == 4:
+		seconds = int(args[3]) + 3
+		minutes = int(args[2])
+		hours = int(args[1])
+		if seconds >= 60:
+			seconds -= 60
+			minutes += 1
+		if minutes == 60:
+			minutes = 0
+			hours += 1
+		time_data = str(hours) + ":" + str(minutes) + ":" + str(seconds)
+		# Set NEXT_TIME for next UDP command
+		NEXT_TIME = datetime.strptime(time_data, "%H:%M:%S")
+		# If scheduled time is behind current time, we set NEXT_TIME = None
+		if NEXT_TIME.time <= datetime.now().time():
+			_logger.info("Timestamp set failed.")
+			NEXT_TIME = None
+			return
+		_logger.info("Next action will be done in %s", time_data)
+		return
+	
+	# UDP Command example: /PVM <action_name> <action_value>
 	command = args[1]
```

Comment:
as is it right now, `args[1]` == `HH`, remember that the format now is `/PVM HH MM SS <action_name> <action_value>`

Reply:
Done